runtime-modules/project-token/src/lib.rs

```diff
@@ -1619,6 +1853,11 @@ impl<T: Config> Module<T> {
         <T as Config>::ModuleId::get().into_sub_account_truncating(Vec::<u8>::new())
     }
 
+    /// Returns teh account for the bonding curve treasury
+    pub fn module_bonding_curve_reserve_account(token_id: T::TokenId) -> T::AccountId {
+        <T as Config>::ModuleId::get().into_sub_account_truncating(&("BONDING_RESERVE", token_id))
```

Comment:
It's risky to use long strings here (longer than 10-12 bytes), because `into_sub_account_truncating` will:
- Add a prefix which is 12 bytes
- Concat it with your input (`BONDING_RESERVE` string itself is already 15 bytes)
- Turncate the result to match `AccountId` length (which is 32 bytes)

If the input is too long, an important part of it (like `token_id` in this case) may get turncated and this will cause huge issues.

How about:
```suggestion
        <T as Config>::ModuleId::get().into_sub_account_truncating(&("AMM", token_id))
```